src/components/config-tabs/tab-contents/DenominatorRelations.js

```diff
@@ -7,20 +7,171 @@ import {
+                    action="Delete"
+                    destructive
+                    onClose={closeModal}
+                    onConfirm={deleteRelation}
+                />
+            )}
+        </>
+    )
+}
+DeleteRelationButton.propTypes = {
+    relation: PropTypes.object,
+}
+
+export const DenominatorRelations = ({ toggleState }) => {
+    const configurations = useConfigurations()
+    const relations = configurations.denominatorRelations
+    const dispatch = useConfigurationsDispatch()
+
+    const [isModalOpen, setIsModalOpen] = useState(false)
+    const [newDenominatorRelationInfo, setNewDenominatorRelationInfo] =
+        useState({
+            A: '',
+            B: '',
+            code: '',
+            criteria: 10,
+            name: '',
+            type: '',
+        })
+    const [selectedDenominatorA, setSelectedDenominatorA] = useState('')
+    const [selectedDenominatorB, setSelectedDenominatorB] = useState('')
+    const [filteredDenominators, setFilteredDenominators] = useState([])
+    const [updateType, setUpdateType] = useState('')
+    const openModal = useCallback(() => setIsModalOpen(true), [])
+    const closeModal = useCallback(() => setIsModalOpen(false), [])
+
+    const handleDenominatorTypeChange = (type) => {
+        // reset the selected denominators
+        setSelectedDenominatorA('')
+        setSelectedDenominatorB('')
+
+        // filter denominators by type
+        setFilteredDenominators(
+            filterDenominatorsByType(configurations.denominators, type)
+        )
+        setNewDenominatorRelationInfo({
+            ...newDenominatorRelationInfo,
+            type: type,
+        })
+    }
+
+    // TODO: will put the edit and add denominators in different components to avoid if conditions below
+    const onSaveDenominatorUpdates = useCallback((updateType) => {
+        if (updateType === 'create') {
+            dispatch({
+                type: CREATE_DENOMINATOR_RELATION,
+                payload: { newDenominatorRelationInfo },
+            }),
+                [dispatch]
+        } else if (updateType === 'update') {
+            dispatch({
+                type: UPDATE_DENOMINATOR_RELATION,
+                payload: {
+                    code: newDenominatorRelationInfo.code,
+                    updatedDenominatorRelation: newDenominatorRelationInfo,
+                },
+            })[(dispatch, newDenominatorRelationInfo.code)]
+        }
+
+        closeModal()
+    })
+
+    // while in editing mode
+    const onEditRelation = (relation) => {
+        openModal()
+        // get the filtered relations first and then only set A and B if A and B are in those filtered relations
+        const filteredDenominators = filterDenominatorsByType(
+            configurations.denominators,
+            relation.type
+        )
+        setFilteredDenominators(filteredDenominators)
+        setNewDenominatorRelationInfo(relation)
+        setSelectedDenominatorA(
```

Comment:
Here you're setting the state variable to be the entire denominator, rather than just the code, so these won't appear in the select when you open the edit.

You could instead do:

```
        setSelectedDenominatorA(
            filteredDenominators.find(
                (denominator) => denominator.code === relation.A
            )?.code
        )
```

or slightly refactor

Reply:
fixed